x-pack/plugins/observability/public/pages/slos/components/slo_list_item.tsx

```diff
@@ -92,12 +96,19 @@ export function SloListItem({
     queryClient.invalidateQueries(['fetchRulesForSlo']);
   };
 
-  const handleNavigateToRules = () => {
-    navigateToUrl(
-      basePath.prepend(
-        `${paths.observability.rules}?_a=(lastResponse:!(),search:%27%27,params:(sloId:%27${slo?.id}%27),status:!(),type:!())`
-      )
-    );
+  const handleNavigateToRules = async () => {
+    const locator = locators.get<RulesParams>(rulesLocatorID);
+
+    if (slo?.id) {
```

Comment:
Is this change of logic intentional? (If slo?.id in undefined, then nothing happens) How probable is this scenario?

In another scenario, if ruleId wasn't available, we redirected the user to the rules page instead. (But maybe that's not a clear option too) 
Alternatives are:
- showing a toast message that we don't have sloId and something is wrong
- maybe not showing the menu item related to this navigation in the first place

cc @maciejforcone

Reply:
@maryam-saeidi  This case can't happen. Removed the if clause.